src/main/clojure/clara/rules/compiler.clj

```diff
@@ -235,15 +235,21 @@
   "Compiles a sequence of constraints into a structure that can be evaluated.
 
    Callers may also pass a collection of equality-only-variables, which instructs
-   this function to only do an equality check on them rather than create a unification binding."
+   this function to only do an equality check on them rather than create a unification binding.
+
+   The returned expression evaluates to a two item vector, [bindings error], which contains
+   the evaluated bindings in the first argument and a map with information on any exception
+   that might have been thrown by evaluating a constraint as the second vector entry."
```

Comment:
I'm going to push soon. Unfortunately, I'm generating a bunch of ClojureScript dead code warnings when compiling tests because the compiler has picked up that ClojureScript expressions such as  `(< degrees 20)` cannot throw. However, `(< nil 20)` does throw in Clojure. There is no correctness issue, but it does pollute the output of `lein compile`.